info.go

```diff
@@ -499,9 +499,38 @@ func newProgramInfoFromFd(fd *sys.FD) (*ProgramInfo, error) {
 		}
 	}
 
+	if len(pi.Name) == len(info.Name)-1 { // Possibly truncated, check BTF info for full name
+		name, err := readNameFromFunc(&pi)
+		if err == nil {
+			pi.Name = name
+		} // If an error occurs, keep the truncated name, which is better than none
+	}
+
 	return &pi, nil
 }
 
+func readNameFromFunc(pi *ProgramInfo) (string, error) {
+	if pi.numFuncInfos == 0 {
+		return "", errors.New("no function info")
+	}
+
+	spec, err := pi.btfSpec()
+	if err != nil {
+		return "", err
+	}
+
+	funcInfos, err := btf.LoadFuncInfos(
+		bytes.NewReader(pi.funcInfos),
+		internal.NativeEndian,
+		1, // Only the first info
+		spec,
+	)
+	if err != nil {
+		return "", err
+	}
+	return funcInfos[0].Func.Name, nil
```

Comment:
This code a bad assumption. That the first function info in the BTF is the func info for the entrypoint, but this is not guaranteed.

Additionally, the code currently assumes `funcInfos` always has at least 1 element, because `pi.numFuncInfos` != 0. But if something when silently wrong somewhere, `funcInfos` could have 0 elements, and the unchecked access at index 0 causes a panic. 

The proper way to do this is to request all func infos (`pi.numFuncInfos`), so that we read back all func infos from the kernel. Then loop over the funcInfos until you find the function with `Offset` 0. It should be the first entry, since `btf.LoadFuncInfos` sorts by `Offset`, but we don't have to depend on that behavior. Bonus points, the loop also means that if we have 0 elements, we don't panic.